app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,97 @@
+class TasksController < ApplicationController
+  
+  def index
+    @tasks = Task.all
+    
+  end
+  
+  def show
+    task = params[:id]
+    @task = Task.find_by(id: task)
+    if @task.nil?
+      flash[:error] = "Could not find task with id: #{task}"
```

Comment:
You probably weren't able to use `flash` meaningfully-- we will learn about `flash` so very very soon!